core/matrix/csr.cpp

```diff
@@ -441,7 +456,7 @@ void Csr<ValueType, IndexType>::read(device_mat_data&& data)
     auto size = data.get_size();
     auto exec = this->get_executor();
     auto arrays = data.empty_out();
-    this->row_ptrs_.resize_and_reset(size[0] + 1);
+    this->row_ptrs_ = array<IndexType>{exec, size[0] + 1};
```

Comment:
resize and reset is the better option here as it does not resize if the array was of correct size.

Reply:
If we want `read` to never turn a view into an owning object, we should keep `resize_and_reset`, otherwise this seems fine? It's definitely a slight pessimization performance-wise.

Reply:
This is changed to make `row_ptrs` owning. Since `col_idxs` and `values` will be owning, I thought this would be more consistent. It has worse performance if the same matrix is read multiple times into. For now, I don't think this is urgent to fix.

Reply:
Somehow I forgot that `device_matrix_data` can also be views. In that case, I will revert this change.